backend/cfg/cfg_irc.ml

```diff
@@ -538,20 +543,20 @@ let rec main : round:int -> State.t -> Cfg_with_layout.t -> liveness =
   assign_colors state cfg_with_layout;
   State.invariant state;
   match State.spilled_nodes state with
-  | [] ->
-    if irc_debug then log ~indent:1 "(end of main)";
-    liveness
+  | [] -> if irc_debug then log ~indent:1 "(end of main)"
   | _ :: _ as spilled_nodes ->
     if irc_debug
     then
       List.iter spilled_nodes ~f:(fun reg ->
           log ~indent:1 "/!\\ register %a needs to be spilled" Printmach.reg reg);
-    rewrite state cfg_with_layout spilled_nodes ~reset:true;
+    rewrite state cfg_with_liveness spilled_nodes ~reset:true;
     State.invariant state;
-    main ~round:(succ round) state cfg_with_layout
+    Cfg_with_liveness.invalidate_liveness cfg_with_liveness;
```

Comment:
Why not invalidating liveness inside `rewrite`?

Reply:
It could be invalidated in `rewrite`. I think I wanted
to make it clear that _(i)_ it should not happen when
adding an instruction (but basically once the rewrite
is done) and that _(ii)_ not all calls to `rewrite` imply
an invalidation (but that could be done conditionally
in `rewrite`).